client/lib/after-layout-flush/index.js

```diff
@@ -42,18 +52,20 @@ export default function afterLayoutFlush( func ) {
 		timeoutHandle = setTimeout( () => {
 			// clear the handle to signal that the timeout handler has been executed
 			timeoutHandle = undefined;
-			timeoutFunc( ...args );
+			timeoutFunc( savedArgs );
```

Comment:
The `savedArgs` should be reset back to `undefined` after a function cal is made. Otherwise, there will be a dangling reference to the args that prevents them to be garbage collected.

A very nice premium feature would be also to preserve `this` when wrapping a function with `afterLayoutFlush`. `this` should have the wrapped function is a class method:
```js
MyClass.prototype.checkScroll = afterLayoutFlush( function() {
  // is this pointing to the class instance?
  this.scrollPosition = this.el.offsetTop;
} );

const inst = new MyClass();
inst.checkScroll();
```

See how [lodash debounce](https://github.com/lodash/lodash/blob/master/debounce.js#L92-L100) does it. They call the stored call params `lastThis` and `lastArgs` (slightly better than `savedArgs` imho). Also note how they are careful enough to reset `lastThis` and `lastArgs` even when the wrapped function throws an exception.

Reply:
Change applied, thanks @jsnajdr !

Args should now be garbage collected sooner (when the timeout function runs, rather than when the returned function gets garbage collected), and `this` should be preserved.